### PR TITLE
Resolve TravisCI build errors for C files

### PR DESCRIPTION
**Pull request by Mithilan16**
**Fixes issue:**
This pull request partially resolves issue #3178.  

**Changes:**
For all C files, removed all TravisCI build log errors regarding unused parameters, and errors for functions returning end of non void function with out returning anything

### CODE REVIEW

**Comment by InfiniteCoder** [comment]
Instead of using the install command for libsdl, add it to package.

File: code/artificial_intelligence/src/tsp.c [hunk 1]
```diff
@@ -96,3 +69,25 @@ void nearest_n(int city)
 	printf("1\n");
 	sum += a[u][1];
 }
+
+int main()
```
Comment (arnavb) [inline code/artificial_intelligence/src/tsp.c]: Move all return types to their own line (C Style Guide)

File: code/bit_manipulation/src/magic_number/magic_number.c [hunk 2]
```diff
@@ -19,7 +19,7 @@ int magicNumber(int n, int fac)
     return answer;
 }
 
-int main(int argc, char *argv[])
+int main()
```
Comment (arnavb) [inline code/bit_manipulation/src/magic_number/magic_number.c]: Same here

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**